source/common/json/json_loader.cc

```diff
@@ -1,178 +1,200 @@
 #include "json_loader.h"
 
-// Do not let jansson leak outside of this file.
+// Do not let RapidJson leak outside of this file.
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
-#include "jansson.h"
+#include "rapidjson/document.h"
+#include "rapidjson/error/en.h"
+#include "rapidjson/istreamwrapper.h"
 #pragma GCC diagnostic pop
 
 namespace Json {
 
-FileLoader::FileLoader(const std::string& file_path) {
-  json_error_t error;
-  json_ = json_load_file(file_path.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {} line: {} column: {}", error.text,
-                                error.line, error.column));
-  }
-}
-
-FileLoader::~FileLoader() { json_decref(json_); }
-
-StringLoader::StringLoader(const std::string& json) {
-  json_error_t error;
-  json_ = json_loads(json.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {}", error.text));
-  }
-}
-
-StringLoader::~StringLoader() { json_decref(json_); }
-
-Object::EmptyObject Object::empty_;
+/**
+ * Implementation of AbstractObject.
+ */
+class AbstractObjectImplBase : public AbstractObject {
+public:
+  AbstractObjectImplBase(const rapidjson::Value& value, const std::string& name)
+      : name_(name), value_(value) {}
+
+  std::vector<AbstractObjectPtr> asObjectArray() const override {
+    if (!value_.IsArray()) {
+      throw Exception(fmt::format("'{}' is not an array", name_));
+    }
 
-Object::EmptyObject::EmptyObject() : json_(json_object()) {}
+    std::vector<AbstractObjectPtr> object_array;
+    object_array.reserve(value_.Size());
 
-Object::EmptyObject::~EmptyObject() { json_decref(json_); }
+    for (rapidjson::SizeType i = 0; i < value_.Size(); i++) {
```

Comment:
Shoot that was the one I forgot to fix. Fixing now...  And will fix the rest of the for loops.